src/mlpack/methods/ann/layer/mean_pooling.hpp

```diff
@@ -174,11 +185,18 @@ class MeanPooling
         if (colEnd > input.n_cols - 1)
           colEnd = input.n_cols - 1;
 
-        arma::mat subInput = input(
-            arma::span(rowidx, rowEnd),
-            arma::span(colidx, colEnd));
+        size_t kernalArea = (rowEnd - rowidx + 1) * (colEnd - colidx + 1);
```

Comment:
```suggestion
        const size_t kernalArea = (rowEnd - rowidx + 1) * (colEnd - colidx + 1);
```

Just to be verbose on what parameter changes over time and which does not..